webapi/src/main/java/org/sakaiproject/webapi/controllers/CalendarController.java

```diff
@@ -97,40 +75,41 @@
+    @GetMapping(value = "/users/current/calendar", produces = MediaType.APPLICATION_JSON_VALUE)
+    public Map<String, Object> getCurrentUserCalendar() throws UserNotDefinedException {
 
-		Session session = checkSakaiSession();
+        checkSakaiSession();
 
-        return siteService.getUserSites().stream().map(s -> {
+        Map<String, Object> data = new HashMap<>();
+        data.put("events", calendarService.getFilteredEvents(getBasicFilterOptions()).stream().map(convert).collect(Collectors.toList()));
+        data.put("days", calendarService.getUpcomingDaysLimit());
+        return data;
+    }
 
-            try {
-                return ((List<CalendarEvent>) calendarService.getCalendar(calendarService.calendarReference(s.getId(), "main"))
-                    .getEvents(null, null))
-                    .stream().map(convert).collect(Collectors.toList());
-            } catch (Exception e) {
-                log.error("Failed to get calendar events for site {}. An empty list will be generated", s.getId());
-                return Collections.<CalendarEventRestBean>emptyList();
-            }
-        }).flatMap(Collection::stream).collect(Collectors.toList());
-	}
+    @GetMapping(value = "/sites/{siteId}/calendar", produces = MediaType.APPLICATION_JSON_VALUE)
+    public Map<String, Object> getSiteCalendar(@PathVariable String siteId) throws UserNotDefinedException {
 
-	@GetMapping(value = "/sites/{siteId}/calendar", produces = MediaType.APPLICATION_JSON_VALUE)
-    public List<CalendarEventRestBean> getSiteCalendar(@PathVariable String siteId) throws UserNotDefinedException {
+        checkSakaiSession();
 
-		Session session = checkSakaiSession();
-        try {
-            return ((List<CalendarEvent>) calendarService.getCalendar(calendarService.calendarReference(siteId, "main"))
-                .getEvents(null, null))
-                .stream().map(convert).collect(Collectors.toList());
-        } catch (Exception e) {
-            log.error("Failed to get calendar events for site {}. An empty list will be generated", siteId);
-            return Collections.<CalendarEventRestBean>emptyList();
-        }
-	}
+        Map<EventFilterKey, Object> filterOptions = getBasicFilterOptions();
+        filterOptions.put(EventFilterKey.SITE, siteId);
+
+        Map<String, Object> data = new HashMap<>();
+        data.put("events", calendarService.getFilteredEvents(filterOptions).stream().map(convert).collect(Collectors.toList()));
+        data.put("days", calendarService.getUpcomingDaysLimit());
+        return data;
+    }
+
+    private Map<EventFilterKey, Object> getBasicFilterOptions() {
+
+        Map<EventFilterKey, Object> filterOptions = new HashMap<>();
+        Integer eventsLimit = serverConfigurationService.getInt("webapi.calendar.events_limit", 50);
+        filterOptions.put(EventFilterKey.LIMIT, eventsLimit);
```

Comment:
You could just make this one line.

Reply:
It's too long, imo. That's why I left it.

Reply:
Done, actually. One less local variable.